### ISSUE TITLE
getAccessToken denied on backend bucket when running `tf-wrapper.sh`

### ISSUE THREAD

**Issue by jburos** [terraform-google-modules/terraform-example-foundation#923]
### TL;DR

When following the instructions in the [03-networks-hub-spoke README](https://github.com/terraform-google-modules/terraform-example-foundation/tree/master/3-networks-hub-and-spoke), I get an error on step 9:

```sh
% ./tf-wrapper.sh init shared
[...]
"Permission 'iam.serviceAccounts.getAccessToken' denied on resource (or it may not exist)."
```

This bucket does exist, and if I unset GOOGLE_IMPERSONATE_SERVICE_ACCOUNT, I do not see this error though I do get other errors since my personal account doesn't all have the necessary permissions for this project/step.

### Expected behavior

I expected the `init` stage to complete successfully, or at least to find the backend.

### Observed behavior

I get an error that my service account does not have access to the backend bucket.

```sh
gcp-networks % ./tf-wrapper.sh init shared
refs doesn't match shared; skipping
objects doesn't match shared; skipping
logs doesn't match shared; skipping
info doesn't match shared; skipping
hooks doesn't match shared; skipping
*************** TERRAFORM INIT *******************
      At environment: envs/shared 
**************************************************
Initializing modules...
- base_shared_vpc in ../../modules/base_shared_vpc
Downloading registry.terraform.io/terraform-google-modules/network/google 5.2.0 for base_shared_vpc.main...
- base_shared_vpc.main in .terraform/modules/base_shared_vpc.main
- base_shared_vpc.main.firewall_rules in .terraform/modules/base_shared_vpc.main/modules/firewall-rules
- base_shared_vpc.main.routes in .terraform/modules/base_shared_vpc.main/modules/routes
- base_shared_vpc.main.subnets in .terraform/modules/base_shared_vpc.main/modules/subnets
- base_shared_vpc.main.vpc in .terraform/modules/base_shared_vpc.main/modules/vpc
Downloading registry.terraform.io/terraform-google-modules/network/google 5.2.0 for base_shared_vpc.peering...
- base_shared_vpc.peering in .terraform/modules/base_shared_vpc.peering/modules/network-peering
Downloading registry.terraform.io/terraform-google-modules/cloud-dns/google 3.1.0 for base_shared_vpc.peering_zone...
- base_shared_vpc.peering_zone in .terraform/modules/base_shared_vpc.peering_zone
Downloading registry.terraform.io/terraform-google-modules/network/google 5.2.0 for base_shared_vpc.private_service_connect...
- base_shared_vpc.private_service_connect in .terraform/modules/base_shared_vpc.private_service_connect/modules/private-service-connect
Downloading registry.terraform.io/terraform-google-modules/cloud-dns/google 4.2.0 for base_shared_vpc.private_service_connect.gcr...
- base_shared_vpc.private_service_connect.gcr in .terraform/modules/base_shared_vpc.private_service_connect.gcr
Downloading registry.terraform.io/terraform-google-modules/cloud-dns/google 4.2.0 for base_shared_vpc.private_service_connect.googleapis...
- base_shared_vpc.private_service_connect.googleapis in .terraform/modules/base_shared_vpc.private_service_connect.googleapis
Downloading registry.terraform.io/terraform-google-modules/cloud-dns/google 4.2.0 for base_shared_vpc.private_service_connect.pkg_dev...
- base_shared_vpc.private_service_connect.pkg_dev in .terraform/modules/base_shared_vpc.private_service_connect.pkg_dev
Downloading registry.terraform.io/terraform-google-modules/cloud-router/google 3.0.0 for base_shared_vpc.region1_router1...
- base_shared_vpc.region1_router1 in .terraform/modules/base_shared_vpc.region1_router1
Downloading registry.terraform.io/terraform-google-modules/cloud-router/google 3.0.0 for base_shared_vpc.region1_router2...
- base_shared_vpc.region1_router2 in .terraform/modules/base_shared_vpc.region1_router2
Downloading registry.terraform.io/terraform-google-modules/cloud-router/google 3.0.0 for base_shared_vpc.region2_router1...
- base_shared_vpc.region2_router1 in .terraform/modules/base_shared_vpc.region2_router1
Downloading registry.terraform.io/terraform-google-modules/cloud-router/google 3.0.0 for base_shared_vpc.region2_router2...
- base_shared_vpc.region2_router2 in .terraform/modules/base_shared_vpc.region2_router2
- base_transitivity in ../../modules/transitivity
Downloading registry.terraform.io/GoogleCloudPlatform/lb-internal/google 5.0.0 for base_transitivity.ilbs...
- base_transitivity.ilbs in .terraform/modules/base_transitivity.ilbs
Downloading registry.terraform.io/terraform-google-modules/vm/google 7.9.0 for base_transitivity.migs...
- base_transitivity.migs in .terraform/modules/base_transitivity.migs/modules/mig
Downloading registry.terraform.io/terraform-google-modules/service-accounts/google 4.2.0 for base_transitivity.service_account...
- base_transitivity.service_account in .terraform/modules/base_transitivity.service_account
Downloading registry.terraform.io/terraform-google-modules/vm/google 7.9.0 for base_transitivity.templates...
- base_transitivity.templates in .terraform/modules/base_transitivity.templates/modules/instance_template
Downloading registry.terraform.io/terraform-google-modules/cloud-dns/google 4.2.0 for dns-forwarding-zone...
- dns-forwarding-zone in .terraform/modules/dns-forwarding-zone
Downloading registry.terraform.io/terraform-google-modules/cloud-router/google 3.0.0 for dns_hub_region1_router1...
- dns_hub_region1_router1 in .terraform/modules/dns_hub_region1_router1
Downloading registry.terraform.io/terraform-google-modules/cloud-router/google 3.0.0 for dns_hub_region1_router2...
- dns_hub_region1_router2 in .terraform/modules/dns_hub_region1_router2
Downloading registry.terraform.io/terraform-google-modules/cloud-router/google 3.0.0 for dns_hub_region2_router1...
- dns_hub_region2_router1 in .terraform/modules/dns_hub_region2_router1
Downloading registry.terraform.io/terraform-google-modules/cloud-router/google 3.0.0 for dns_hub_region2_router2...
- dns_hub_region2_router2 in .terraform/modules/dns_hub_region2_router2
Downloading registry.terraform.io/terraform-google-modules/network/google 5.2.0 for dns_hub_vpc...
- dns_hub_vpc in .terraform/modules/dns_hub_vpc
- dns_hub_vpc.firewall_rules in .terraform/modules/dns_hub_vpc/modules/firewall-rules
- dns_hub_vpc.routes in .terraform/modules/dns_hub_vpc/modules/routes
- dns_hub_vpc.subnets in .terraform/modules/dns_hub_vpc/modules/subnets
- dns_hub_vpc.vpc in .terraform/modules/dns_hub_vpc/modules/vpc
- hierarchical_firewall_policy in ../../modules/hierarchical_firewall_policy
- restricted_shared_vpc in ../../modules/restricted_shared_vpc
Downloading registry.terraform.io/terraform-google-modules/vpc-service-controls/google 4.0.1 for restricted_shared_vpc.access_level_members...
- restricted_shared_vpc.access_level_members in .terraform/modules/restricted_shared_vpc.access_level_members/modules/access_level
Downloading registry.terraform.io/terraform-google-modules/network/google 5.2.0 for restricted_shared_vpc.main...
- restricted_shared_vpc.main in .terraform/modules/restricted_shared_vpc.main
- restricted_shared_vpc.main.firewall_rules in .terraform/modules/restricted_shared_vpc.main/modules/firewall-rules
- restricted_shared_vpc.main.routes in .terraform/modules/restricted_shared_vpc.main/modules/routes
- restricted_shared_vpc.main.subnets in .terraform/modules/restricted_shared_vpc.main/modules/subnets
- restricted_shared_vpc.main.vpc in .terraform/modules/restricted_shared_vpc.main/modules/vpc
Downloading registry.terraform.io/terraform-google-modules/network/google 5.2.0 for restricted_shared_vpc.peering...
- restricted_shared_vpc.peering in .terraform/modules/restricted_shared_vpc.peering/modules/network-peering
Downloading registry.terraform.io/terraform-google-modules/cloud-dns/google 3.1.0 for restricted_shared_vpc.peering_zone...
- restricted_shared_vpc.peering_zone in .terraform/modules/restricted_shared_vpc.peering_zone
Downloading registry.terraform.io/terraform-google-modules/network/google 5.2.0 for restricted_shared_vpc.private_service_connect...
- restricted_shared_vpc.private_service_connect in .terraform/modules/restricted_shared_vpc.private_service_connect/modules/private-service-connect
Downloading registry.terraform.io/terraform-google-modules/cloud-dns/google 4.2.0 for restricted_shared_vpc.private_service_connect.gcr...
- restricted_shared_vpc.private_service_connect.gcr in .terraform/modules/restricted_shared_vpc.private_service_connect.gcr
Downloading registry.terraform.io/terraform-google-modules/cloud-dns/google 4.2.0 for restricted_shared_vpc.private_service_connect.googleapis...
- restricted_shared_vpc.private_service_connect.googleapis in .terraform/modules/restricted_shared_vpc.private_service_connect.googleapis
Downloading registry.terraform.io/terraform-google-modules/cloud-dns/google 4.2.0 for restricted_shared_vpc.private_service_connect.pkg_dev...
- restricted_shared_vpc.private_service_connect.pkg_dev in .terraform/modules/restricted_shared_vpc.private_service_connect.pkg_dev
Downloading registry.terraform.io/terraform-google-modules/cloud-router/google 3.0.0 for restricted_shared_vpc.region1_router1...
- restricted_shared_vpc.region1_router1 in .terraform/modules/restricted_shared_vpc.region1_router1
Downloading registry.terraform.io/terraform-google-modules/cloud-router/google 3.0.0 for restricted_shared_vpc.region1_router2...
- restricted_shared_vpc.region1_router2 in .terraform/modules/restricted_shared_vpc.region1_router2
Downloading registry.terraform.io/terraform-google-modules/cloud-router/google 3.0.0 for restricted_shared_vpc.region2_router1...
- restricted_shared_vpc.region2_router1 in .terraform/modules/restricted_shared_vpc.region2_router1
Downloading registry.terraform.io/terraform-google-modules/cloud-router/google 3.0.0 for restricted_shared_vpc.region2_router2...
- restricted_shared_vpc.region2_router2 in .terraform/modules/restricted_shared_vpc.region2_router2
Downloading registry.terraform.io/terraform-google-modules/vpc-service-controls/google 4.0.1 for restricted_shared_vpc.regular_service_perimeter...
- restricted_shared_vpc.regular_service_perimeter in .terraform/modules/restricted_shared_vpc.regular_service_perimeter/modules/regular_service_perimeter
- restricted_transitivity in ../../modules/transitivity
Downloading registry.terraform.io/GoogleCloudPlatform/lb-internal/google 5.0.0 for restricted_transitivity.ilbs...
- restricted_transitivity.ilbs in .terraform/modules/restricted_transitivity.ilbs
Downloading registry.terraform.io/terraform-google-modules/vm/google 7.9.0 for restricted_transitivity.migs...
- restricted_transitivity.migs in .terraform/modules/restricted_transitivity.migs/modules/mig
Downloading registry.terraform.io/terraform-google-modules/service-accounts/google 4.2.0 for restricted_transitivity.service_account...
- restricted_transitivity.service_account in .terraform/modules/restricted_transitivity.service_account
Downloading registry.terraform.io/terraform-google-modules/vm/google 7.9.0 for restricted_transitivity.templates...
- restricted_transitivity.templates in .terraform/modules/restricted_transitivity.templates/modules/instance_template

Initializing the backend...
╷
│ Error: Failed to get existing workspaces: querying Cloud Storage failed: Get "https://storage.googleapis.com/storage/v1/b/bkt-prj-bs-seed-tfstate-5bb6/o?alt=json&delimiter=%2F&pageToken=&prefix=terraform%2Fnetworks%2Fenvs%2Fshared%2F&prettyPrint=false&projection=full&versions=false": impersonate: status code 403: {
│   "error": {
│     "code": 403,
│     "message": "Permission 'iam.serviceAccounts.getAccessToken' denied on resource (or it may not exist).",
│     "status": "PERMISSION_DENIED",
│     "details": [
│       {
│         "@type": "type.googleapis.com/google.rpc.ErrorInfo",
│         "reason": "IAM_PERMISSION_DENIED",
│         "domain": "iam.googleapis.com",
│         "metadata": {
│           "permission": "iam.serviceAccounts.getAccessToken"
│         }
│       }
│     ]
│   }
│ }
│ 
│ 
│ 
╵
```

### Terraform Configuration

```hcl
I changed almost nothing from the template provided here (I'm testing).
```

### Terraform Version

```sh
Terraform v1.3.7
on darwin_amd64
```
```


### Additional information

_No response_

**Comment by jburos** [comment]
Apologies - newb to gcloud here!

In case anyone else comes across this in the future, once I granted the user executing the terraform command (myself) the `roles/iam.serviceAccountUser` role I was able to execute this.

**Comment by mikebridge** [comment]
I also had to set `roles/iam.serviceAccountTokenCreator` for this step.  (And wait a minute for it to propagate)

**Comment by pascalwhoop** [comment]
i still wonder why one runs into this ( me to)

**Comment by Thintalle** [comment]
We had the same issue. The Workaround was successful, too.

Terraform 1.11.4
TEF 4.1.0

We used automatic group creation (which has issues on its own...)
